storage/Devicegraph.cc

```diff
@@ -43,13 +44,15 @@ namespace storage
 
 
     WrongNumberOfParents::WrongNumberOfParents(size_t seen, size_t expected)
-	: Exception(sformat("wrong number of parents, seen '%zu', expected '%zu'", seen, expected))
+	: Exception(sformat("wrong number of parents, seen '%zu', expected '%zu'", seen, expected),
+		    Silencer::is_any_active() ? DEBUG : WARNING)
```

Comment:
OK, it is not so critical. I just see it as example of "tell, not ask" principle. Now you need to ask silencer if it is active so ask...for tell it will be more give me log level. So responsibility to choose silenced log level is on silencer and not user.